lib/src/main/kotlin/com/swmansion/starknet/data/serializers/TransactionPolymorphicSerializer.kt

```diff
@@ -7,14 +7,15 @@ import kotlinx.serialization.json.*
 
 internal object TransactionPolymorphicSerializer : JsonContentPolymorphicSerializer<Transaction>(Transaction::class) {
     override fun selectDeserializer(element: JsonElement): DeserializationStrategy<out Transaction> =
+        // pick deserializer based on transaction type field
         when (element.jsonObject["type"]?.jsonPrimitive?.content) {
             "INVOKE_FUNCTION" -> selectInvokeDeserializer(element)
             "INVOKE" -> selectInvokeDeserializer(element)
             "DECLARE" -> DeclareTransaction.serializer()
             "DEPLOY" -> DeployTransaction.serializer()
             "DEPLOY_ACCOUNT" -> DeployAccountTransaction.serializer()
             "L1_HANDLER" -> L1HandlerTransaction.serializer()
-            else -> throw IllegalArgumentException("Invalid transaction type")
+            else -> throw IllegalArgumentException("Invalid transaction type ${element.jsonObject["type"]?.jsonPrimitive?.content}")
```

Comment:
Please wrap content with `''` or `[]`. In case content = `""` the content won't be visible to user without such wrapping.